<h3> UK ROAD ACCIDENT ANALYSIS</h3>
<hr> Analyst: Nathaniel Esguerra

In [1]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt

In [2]:
uk_road_accident = pd.read_csv('datasets\\uk_road_accident.csv')

In [3]:
uk_road_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h3> 1. Which districts recorded the most accidents?</h3>

In [4]:
uk_road_accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<hr> Insight#1 - Birmingham recorded the most accidents (13,491), significantly higher than Leeds (8,898) and Manchester (6,720).

<h3 2. Which vehicles are most often involved in deadly accidents?

<h3> 2. Which vehicles are most often involved in deadly accidents?</h3>

In [5]:
vehicle_type_df = uk_road_accident['Vehicle_Type']

In [6]:
vehicle_type_df

0                                          Car
1                                          Car
2                        Taxi/Private hire car
3         Bus or coach (17 or more pass seats)
4                                Other vehicle
                          ...                 
660674                                     Car
660675                                     Car
660676                                     Car
660677                   Motorcycle over 500cc
660678                                     Car
Name: Vehicle_Type, Length: 660679, dtype: object

<hr> Insight#2 - Cars are the most involved vehicle type in deadly accidents, with a total of 660,679 cases

<h3> 3. What is the mean number of casualties in each accident? </h3>

In [7]:
uk_road_accident['Number_of_Casualties'].mean()

np.float64(1.357040257068864)

<hr> Insight#3 - The majority of accidents fall under the 'Slight' severity category.

<h3> 4. How many accidents happened in urban vs. rural areas?</h3>

In [8]:
fatal = uk_road_accident[uk_road_accident['Accident_Severity'] == 'Fatal']

In [9]:
fatal

Index Accident_Severity Accident Date   Latitude  \
7       200701BS67370             Fatal     3/10/2019  51.482260   
11      200701BS67747             Fatal     2/11/2019  51.499127   
20      200701BS68301             Fatal    24-10-2019  51.505996   
21      200701BS68302             Fatal    24-11-2019  51.486795   
717     200701CP00032             Fatal    30-01-2019  51.510978   
...               ...               ...           ...        ...   
660470  201091NH00529             Fatal    29-01-2022  58.475726   
660519  201091NJ01533             Fatal      2/2/2022  58.495554   
660565  201091NJ11329             Fatal    15-08-2022  57.871514   
660581  201091NJ14695             Fatal    21-10-2022  58.445475   
660627  201091NL00060             Fatal      6/1/2022  60.241195   

             Light_Conditions           District Area  Longitude  \
7       Darkness - lights lit  Kensington and Chelsea  -0.179486   
11                   Daylight  Kensington and Chelsea  -0.159652   
20                   Daylight  Kensington and Chelsea  -0.214277   
21                   Daylight  Kensington and Chelsea  -0.170375   
717                  Daylight          City of London  -0.107581   
...                       ...                     ...        ...   
660470               Daylight           Western Isles  -6.278583   
660519               Daylight                Highland  -4.451628   
660565               Daylight                Highland  -4.243081   
660581  Darkness - lights lit                Highland  -3.065535   
660627               Daylight        Shetland Islands  -1.409761   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
7                          3                   2                     Dry   
11                         1                   2                     Dry   
20                         1                   1                     Dry   
21                         5                   1                     Dry   
717                        1                   2             Wet or damp   
...                      ...                 ...                     ...   
660470                     1                   1             Wet or damp   
660519                     1                   1                    Snow   
660565                     1                   1                     Dry   
660581                     1                   1             Wet or damp   
660627                     3                   1                    Snow   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
7       Single carriageway               Urban     Fine no high winds   
11      Single carriageway               Urban     Fine no high winds   
20      Single carriageway               Urban     Fine no high winds   
21      Single carriageway               Urban     Fine no high winds   
717       Dual carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660470  Single carriageway               Rural      Fine + high winds   
660519  Single carriageway               Rural  Snowing no high winds   
660565  Single carriageway               Rural     Fine no high winds   
660581  Single carriageway               Rural     Fine no high winds   
660627  Single carriageway               Rural     Fine no high winds   

                                Vehicle_Type  
7                                        Car  
11      Bus or coach (17 or more pass seats)  
20                                       Car  
21                                       Car  
717                                      Car  
...                                      ...  
660470                                   Car  
660519                                   Car  
660565         Goods 7.5 tonnes mgw and over  
660581                                   Car  
660627   Van / Goods 3.5 tonnes mgw or under  

[8661 rows x 14 columns]

<hr> Insight#4 - Based on the dataset, a total of 6,813 accidents occurred in urban areas, while 1,848 accidents were recorded in rural areas.

<h3> 5. Which road type is most frequently associated with accidents? </h3>

In [10]:
uk_road_accident['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<hr> Insight#5 - The dataset shows that most accidents happened on dry roads (447,821 cases), followed by wet or damp roads (186,708 cases). Other conditions like frost or ice (18,517), snow (5,890), and flooding (1,017) were much less common.

<h3> 6. What weather condition is most common during accidents?</h3>

In [13]:
uk_road_accident['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<hr> Insight#6 - Most accidents happened in fine weather with no high winds (520,885 cases), followed by rain without high winds (79,696 cases). Other conditions like fog, snow, or strong winds were much less common.

<h3> 7. How many accidents happened during daylight vs. darkness?</h3>

uk_road_accident['Light_Conditions'].value_counts()

<hr> Insight#7 - Most accidents happened in daylight (484,880 cases), while a total of 175,799 accidents occurred under different darkness conditions.

<h3> 8. How many total accidents are recorded?</h3>

In [15]:
len(uk_road_accident)

660679

<hr> Insight#8 - As what indicated above, the total accidents recorded was 660,679.

<h3> 9. How many accidents involved motorcycles?</h3>

In [16]:
motorcycle_accidents = uk_road_accident[uk_road_accident['Vehicle_Type'].str.contains('Motorcycle', case=False, na=False)]
len(motorcycle_accidents)

56185

<hr> Insight#9 - As what indicated, the total accident recorded was 56185.

<h3> 10. How many accidents occurred when the road surface was dry?

In [17]:
dry_accidents = uk_road_accident[uk_road_accident['Road_Surface_Conditions'] == 'Dry']
len(dry_accidents)

447821

<hr> Insight#10 - Based on the dataset, a total of 447,821 accidents occurred on dry road surfaces.

<h3> 11. Which weather condition is the least common?</h3>

In [18]:
uk_road_accident['Weather_Conditions'].value_counts()


Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<hr> Insight#11 - The least common weather condition during accidents is snow with high winds, with 885 cases.

<h3> 12. How many accidents happened in darkness with no lighting?</h3>

In [19]:
dark_no_light_accidents = uk_road_accident[uk_road_accident['Light_Conditions'] == 'Darkness - no lighting']
len(dark_no_light_accidents)

37437

<hr> Insight#12 - Based on the information, a total of 37,437 accidents occurred in darkness with no lighting.

<h3> 13. What is the latest accident date in the dataset?</h3>

In [20]:
uk_road_accident['Accident Date'].max()

'9/9/2022'

<hr> Insight#13 - The latest accident in the dataset happened on September 9, 2022.

<h3> 14. What is the earliest accident date in the dataset?</h3>

In [21]:
uk_road_accident['Accident Date'].min()

'1/1/2019'

<hr> Insight#14 - The earliest accident in the dataset happened on January 1, 2019.

<h3> 15. Which district area had the fewest accidents?</h3>

In [23]:
uk_road_accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<hr> Insight#15 - The district area with the fewest accidents is Clackmannanshire, with 91 recorded cases.

<h3> 16. How many accidents happened in Highland?</h3>

In [24]:
highland_accidents = uk_road_accident[uk_road_accident['District Area'] == 'Highland']
len(highland_accidents)

2021

<hr> Insight#16 - A total of 2,021 accidents happened in Highland.

<h3> 17. Which district area had the most serious accidents? </h3>

In [26]:
serious_accidents = uk_road_accident[uk_road_accident['Accident_Severity'] == 'Serious']
serious_accidents['District Area'].value_counts().idxmax()

'Birmingham'

<hr> Insight#17 - The district with the most serious accidents is Birmingham.

<h3> 18. How many accidents happen on wet roads?</h3>

In [27]:
wet_road_accidents = uk_road_accident[uk_road_accident['Road_Surface_Conditions'] == 'Wet or damp']
len(wet_road_accidents)

186708

<hr> Insight#18 - A total of 186,708 accidents occurred on wet roads.

<h3> 19. How many accidents happen on dry roads?</h3>

In [28]:
dry_road_accidents = uk_road_accident[uk_road_accident['Road_Surface_Conditions'] == 'Dry']
len(dry_road_accidents)


447821

<hr> Insight#19 - A total of 447,821 accidents occurred on dry roads.

<h3> 20.  Is there a correlation between latitude and longitude and the number of casualties?</h3>

In [29]:
uk_road_accident[['Latitude', 'Longitude', 'Number_of_Casualties']].corr()

Latitude  Longitude  Number_of_Casualties
Latitude              1.000000  -0.398114              0.032201
Longitude            -0.398114   1.000000             -0.040406
Number_of_Casualties  0.032201  -0.040406              1.000000